### DISCUSSION TITLE
Does Keka have it's own icons for the file types/extensions it associates with?

### DISCUSSION THREAD

**Discussion by navinthomas** [aonez/Keka#1105] [Q&A] [answered]
I used to use BetterZip and the Unarchiver prior to using Keka. They are still kept installed for backup reasons, and BetterZip has a great Finder preview extension.

I have used the Keka helper app to set Keka as the default extraction application. And yet, all the file icons or archive files come from either BetterZip or the Unarchiver.

Is this the normal behaviour? Does Keka not have its own file icons for supported file types?

**Comment by aonez** [comment]
Keka does have icons for all its supported file types. Since Catalina (if I remember correctly) the handling of the icons and some file metadata changed and this kind of errors started appearing. For example https://github.com/aonez/Keka/issues/453 and https://github.com/aonez/Keka/issues/472.

Take a look at the "Open With" option in the context menu of a file compatible with Keka. There the first option (stated as Default) should be Keka if it was set as the default properly. Also in the "Get Info" panel you can check and also change the default app.

**Reply by navinthomas** [reply]
Keka is already the default app. I confirmed with Get Info and Open With. But still the icons look like BetterZip icons and Unarchiver icons

**Reply by navinthomas** [reply]
I'm using macOS Monterey (12.3.1).

**Reply by navinthomas** [reply]
@aonez If the issue is fixed in Big Sur already, and I am using Monterey, does that mean that what I am facing is a completely different issue all together?

**Reply by aonez** [reply]
Not sure here. If you create another user account in that Mac, do you still see mixed icons? Also try (temporarily) removing BetterZip and The Unarchiver and see if that (with a proper restart) helps.

We used in the past a code to cleanup the icon caches in the system, if you want to try this too:
```
sudo rm -rfv /Library/Caches/com.apple.iconservices.store; sudo find /private/var/folders/ \( -name com.apple.dock.iconcache -or -name com.apple.iconservices \) -exec rm -rfv {} \; ; sleep 3; killall Dock; killall Finder
```

**Reply by navinthomas** [reply]
@aonez 
Earlier, RAR files had a BetterZip icon, while ZIP files had an Unarchiver icon. After removing both of those apps and restarting the system, the file icons have at least reverted to the default macOS icons for those files.

![image](https://user-images.githubusercontent.com/34918736/170530759-c44a69e9-ea42-4253-87ab-d4fb070642fa.png) ![image](https://user-images.githubusercontent.com/34918736/170530817-4641130e-7dd7-42be-8737-4fabe01e138d.png)

As you can see, Keka's file icons are not getting associated with those archives, even though it is still the default app for them. Running the command in terminal did not have any noticeable effect on the system. I reinstall BetterZip to get it's quick look feature, and now the RAR file icons have become BeterZip's icons again.

So, is Keka simply unable to associate the necessary file icons for archives in my system? Is it something that is done by the helper app?

**Reply by aonez** [reply]
The purple icon in the RAR file you’re sharing is from Keka, are you seeing this one now?

**Reply by aonez** [reply]
The helper is only used to set Keka as the default app.

**Reply by navinthomas** [reply]
@aonez 
The purple icon is no longer visible since I reinstalled BetterZip. RAR files now have BetterZip's RAR icon. ZIP file icons still remain as the same one shown in the screenshot above. I believe that comes from the macOS default archive utility app.

Note: I still have Keka as the default app for all archives.

I have not reinstalled the Unarchiver since I do not need it any more. If I do install it, I suspect it will replace the ZIP file icons. Not sure. Just a guess.

**Reply by aonez** [reply]
If you create a new user account, what icon appear in the RAR file?

**Reply by navinthomas** [reply]
@aonez I created a new account and this is what I have observed

**New account**
RAR file - BetterZip icon
ISO file - macOS's icon (same as that for DMG files)
ZIP file - macOS's icon

**Current/old account**
RAR file - BetterZip icon
ISO file - BetterZip icon
ZIP file - macOS's icon

**Comment by navinthomas** [comment, marked as answer]
@aonez I was able to fix this issue by following the steps mentioned in this article -- [How to Clear Icon Cache on Mac](https://osxdaily.com/2022/05/23/clear-icon-cache-mac/).
The step involving Safe Mode was not needed in my case, but I did it anyway.
